.github/workflows/ci.yml

```diff
@@ -227,68 +230,132 @@ jobs:
-        uses: ansys/actions/doc-deploy-dev@v6
-        with:
-          cname: ${{ env.DOCUMENTATION_CNAME }}
-          token: ${{ secrets.GITHUB_TOKEN }}
-          doc-artifact-name: "documentation-html"
-
-  release:
-    name: Release project
-    if: github.event_name == 'push' && contains(github.ref, 'refs/tags')
-    needs: [test, docs]
-    runs-on: ubuntu-latest
+  build-gallery:
+    name: Build Sphinx-Gallery Examples
+    runs-on: public-ubuntu-latest-8-cores
+    needs: [doc-style, build]
     steps:
-      - name: Release to the private PyPI repository
-        uses: ansys/actions/release-pypi-private@v6
+      - uses: actions/checkout@v4
+      - name: Setup Python
+        uses: actions/setup-python@v5
         with:
-          library-name: ${{ env.PACKAGE_NAME }}
-          twine-username: "__token__"
-          twine-token: ${{ secrets.PYANSYS_PYPI_PRIVATE_PAT }}
+          python-version: ${{ env.MAIN_PYTHON_VERSION }}
 
-      - name: "Release to the public PyPI repository"
-        uses: ansys/actions/release-pypi-public@v6
+      - name: Download package
+        uses: actions/download-artifact@v4
         with:
-          library-name: ${{ env.PACKAGE_NAME }}
-          twine-username: "__token__"
-          twine-token: ${{ secrets.PYPI_TOKEN }}
+          name: ${{ env.PACKAGE_NAME }}-artifacts
+          path: dist
 
-      # TODO: We can't use ansys/actions/release-github as it assumes PDF doc.
-      - name: Set up Python
-        uses: actions/setup-python@v5
+      - name: Install pysystemcoupling with doc dependencies
+        run: |
+          wheel_name=`echo dist/*.whl`
+          pip install -q --force-reinstall ${wheel_name}[doc] > /dev/null
+
+      - name: Docker Login
+        uses: docker/login-action@v3
         with:
-          python-version: ${{ env.MAIN_PYTHON_VERSION }}
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
 
-      - uses: actions/checkout@v4
-      - uses: actions/download-artifact@v4
+      - name: Pull Required Docker Images
+        run: |
+          echo "Pulling pysystem-coupling Image Version ${SYC_IMAGE_VERSION}"
+          echo "Pulling pyfluent Image Version ${FLUENT_IMAGE_VERSION}"
+          echo "Pulling mapdl Image Version ${MAPDL_IMAGE_VERSION}"
+          
+          docker pull ghcr.io/ansys/pysystem-coupling:${SYC_IMAGE_VERSION} &
+          docker pull ghcr.io/ansys/pyfluent:${FLUENT_IMAGE_VERSION} &
+          docker pull ghcr.io/ansys/mapdl:${MAPDL_IMAGE_VERSION} &
+          
+          # Wait for the background processes to finish
+          wait
+        env:
+          SYC_IMAGE_VERSION: ${{ env.SYC_IMAGE_VERSION }}
+          FLUENT_IMAGE_VERSION: ${{ env.FLUENT_IMAGE_VERSION }}
+          MAPDL_IMAGE_VESRION: ${{ env.MAPDL_IMAGE_VERSION }}
 
-      - name: Display structure of downloaded files
-        run: ls -R
+      - name: Start MAPDL Container
```

Comment:
Bearing in mind that my understanding of this is vague - I think it's to ensure that the docker network is set up correctly? - I am worried that it might not be scalable to start up the MAPDL container once before the doc build unless it can be reused by multiple examples. I think I would expect container startup normally to be hidden behind launch commands.

Reply:
That's what I would have wanted, but MAPDL does not offer that (as far as I found) - see the docs: https://mapdl.docs.pyansys.com/version/stable/getting_started/launcher.html#connect-to-the-local-mapdl-instance.

So no, I don't love this either, but it seems to be necessary.

Reply:
Sorry, maybe I used the wrong word, but by scalable what I was asking about was whether, if we had multiple examples using MAPDL, the same container that is being started here could be reused across those examples.

Reply:
@ansMHanmer I have been having a look in the pymapdl repo. I notice that when they start the mapdl container for doc builds in their `ci.yml` they specify a `--restart =always` argument to `docker run`. So I guess either the container is reused across examples or it restarts automatically once the first one launched has exited.

Reply:
@ansMHanmer I haven't made much progress yet on this question of reusing an MAPDL container across multiple examples, as I got sidetracked with a question similar to the one I highlighted for Fluent. Ideally we want the example script to run as-is on the desktop, so ideally we want to have a `launch_mapdl` call in the script. From looking at how `pymapdl` does its own doc build, I think the solution is to set `PYMAPDL_START_INSTANCE=FALSE`. This allows us to call `launch_mapdl` but behind the scenes it will connect to a running instance if there is one.

Reply:
Hey @iboyd-ansys makes sense, although I had trouble and continue to have trouble with using the launch_mapdl() method locally.  I tried the example here: https://mapdl.docs.pyansys.com/version/stable/getting_started/launcher.html#connect-to-the-local-mapdl-instance

but it always fails to connect.  Using the `Mapdl()` method works nicely.  I'll do some investigation to see if I can figure that part out.

Reply:
As discussed in our meeting this was due to an incorrect IP that needed to be passed in, but that is not the case when running on GitHub so that's great.